src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -5,10 +5,40 @@
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_AGE = 18;
+    private static final int MIN_LENGTH_DATA = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new InvalidDataException("User must not be null");
+        }
+        if (user.getLogin() == null) {
+            throw new InvalidDataException("Login mustn't be null");
+        }
+        if (user.getPassword() == null) {
+            throw new InvalidDataException("Password mustn't be null");
+        }
+        if (findUserByLogin(user) != null) {
+            throw new InvalidDataException("User with this login has been already registered");
+        }
+        if (user.getLogin().length() < MIN_LENGTH_DATA) {
+            throw new InvalidDataException("Login must contain at least "
+                    + MIN_LENGTH_DATA + " characters");
+        }
+        if (user.getPassword().length() < MIN_LENGTH_DATA) {
+            throw new InvalidDataException("Password must contain at least "
+                    + MIN_LENGTH_DATA + " characters");
+        }
+        if (user.getAge() < MIN_AGE) {
```

Comment:
According to the checklist, you should not add something that is tied to something that may change over time like max user age. However, it seems that MIN_AGE is a constant that is unlikely to change frequently. Just ensure that this constant does not need to be configurable or changeable in the future.